pandas/indexes/base.py

```diff
@@ -144,6 +144,41 @@ class Index(IndexOpsMixin, StringAccessorMixin, PandasObject):
     def __new__(cls, data=None, dtype=None, copy=False, name=None,
                 fastpath=False, tupleize_cols=True, **kwargs):
 
+        def convert_to_int_index(_data, _copy, _name):
```

Comment:
make this a class method instead (nested functions are harder to read); name it _maybe_convert_to_integer_index

use data, copy, name as arg names